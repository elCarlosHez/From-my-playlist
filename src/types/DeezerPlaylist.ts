export type DeezerPlaylist = {
  id: string;
  link: string;
  title: string;  
}
