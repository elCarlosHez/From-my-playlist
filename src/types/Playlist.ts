export type Playlist = {
  id: string;
  name: string;
  href: string;
}
