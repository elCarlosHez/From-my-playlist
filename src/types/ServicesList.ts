export enum ServicesList {
  Spotify = 'Spotify',
  YTMusic = 'Youtube Music',
  Deezer = 'Deezer'
}
