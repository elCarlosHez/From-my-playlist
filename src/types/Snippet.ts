export type Snippet = {
  title: string | undefined
};
