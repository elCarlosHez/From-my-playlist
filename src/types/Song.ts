// href: "https://api.spotify.com/v1/tracks/3cfOd4CMv2snFaKAnMdnvK"
// id: "3cfOd4CMv2snFaKAnMdnvK"
// is_local: false
export type Song = {
  id: string;
  href: string;
  is_local: boolean;
  name: string;
  artists: Artist[];
  album: Album;
}

export type Artist = {
  id: string;
  name: string;
}

export type Album = {
  id: string;
  images: [];
  name: string;
}
