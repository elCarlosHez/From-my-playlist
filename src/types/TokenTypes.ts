export type TokenMessage = {
  type: string,
  token: string,
};

export const TOKEN_SPOTIFY_TYPE = 'Spotify';
export const TOKEN_YOUTUBE_TYPE = 'Youtube';
export const TOKEN_DEEZER_TYPE = 'Deezer';
