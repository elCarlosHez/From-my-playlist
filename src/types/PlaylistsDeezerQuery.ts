import { DeezerPlaylist } from "./DeezerPlaylist";

export type PlaylistsDeezerQuery = {
  data: DeezerPlaylist[];
}
