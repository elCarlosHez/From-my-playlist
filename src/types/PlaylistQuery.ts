import { Playlist } from './Playlist';

export type PlaylistQuery = {
  href: string;
  items: Playlist[];
  limit: number;
};
