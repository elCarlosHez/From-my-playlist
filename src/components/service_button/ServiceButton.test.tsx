import React from "react";
import { create } from "react-test-renderer";
import { ServicesList } from "../../types/ServicesList";
import { ServiceButton } from "./";

describe("Button component", () => {
  test("it shows the expected text when clicked (testing the wrong way!)", () => {
    const button = create(
      <ServiceButton service={ServicesList.Spotify} onPress={() => {}} />
    );
    expect(button.toJSON()).toMatchSnapshot();
  });
});
