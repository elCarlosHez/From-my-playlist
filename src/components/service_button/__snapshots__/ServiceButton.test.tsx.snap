// Vitest Snapshot v1

exports[`Button component > it shows the expected text when clicked (testing the wrong way!) 1`] = `
<button
  className="h-full w-full py-2"
  onClick={[Function]}
>
  <img
    alt="Spotify"
    className="max-h-16 m-auto"
    src=""
  />
</button>
`;
